include/libtorrent/buffer.hpp

```diff
@@ -75,7 +68,7 @@ class buffer
 
 		if (size == 0) return;
 
-		size = round_up8(size);
+		size = (size + 7) & (~std::size_t(0x7)); //round_up8(size);
```

Comment:
ah, that's a good point. it shouldn't be in an anonymous namespace, either collapsed as you did or in the ``detail`` namespace